travis/script.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+source "${0%/*}/setup.sh"
+
+# Mypy only works with Python 3
```

Comment:
mypy has a `--py2` flag.

Reply:
That's meant for testing python2 code, but it needs python 3 to run. There isn't even a python2 pypi package for mypy. I don't want to go install python 3 for the python 2 CI because that would just complicate setup and the python 2 CI is still needed for running test cases on python 2, but I could add a second mypy call to the python 3 CI (i.e. the python 3 CI calls "mypy ." and "mypy --py2 ."). Do you think there is benefit in this? Would "mypy --py2 ." check anything that isn't already checked by "mypy ."?

Reply:
Well same reason as why you want both python2 and python3 CI, they are similar, but there could be subtle differences that CI helps to catch, e.g. there might be places taking different branches based on python versions.
I'm ok with only doing type checking in python3 CI, just in general you want the same settings. Feel free to ignore this for now.

Reply:
It doesn't pass. I want to land this now, so I just added a TODO.